rskj-core/src/main/java/co/rsk/core/types/bytes/BytesSlice.java

```diff
@@ -128,6 +128,42 @@ default Bytes copyBytes() {
     default BytesSlice slice(int from, int to) {
         return new BytesSliceImpl(this, from, to);
     }
+
```

Comment:
What do you think about convert this class to an abstract class? I think it could give us various advantages: 

- We could make both equals and hashcode methods part of the class and not implemented as static. 
- We can force the usage of ` BoundaryUtils.checkArrayIndexParam(length(), index)`  
- Stop using default methods for defining common implementations and facilitate the testing.

Reply:
I'd rather use `interface` here. I guess it's quite similar to `CharSequence` interface that's being used for chars without concrete implementation of `equals` and `hashCode`